tests/migrations/test_state.py

```diff
@@ -366,6 +366,52 @@ class Meta:
         project_state.add_model(ModelState.from_model(B))
         self.assertEqual(len(project_state.apps.get_models()), 2)
 
+    def test_remove_relations(self):
+        """
+        Test that state remains coherent after relation field/model deletion.
```

Comment:
"remains coherent" is a bit undefined -- it seems you mean ensuring relations are still properly defined.